### PR TITLE
Add latest samsung internet android releases

### PR DESCRIPTION
**Pull request by lukewarlow**
#### Summary

Add samsung internet 24 and 25.

#### Test results and supporting details

Manually tested on my phone, no available release notes that I could find.

### CODE REVIEW

File: browsers/samsunginternet_android.json [hunk 1]
```diff
@@ -273,9 +273,20 @@
         },
         "23.0": {
           "release_date": "2023-10-18",
-          "status": "current",
+          "status": "retired",
           "engine": "Blink",
           "engine_version": "115"
+        },
+        "24.0": {
+          "release_date": "2024-01-25",
```
Comment (lukewarlow) [inline browsers/samsunginternet_android.json:281]: Very rough guess based on the date the APK for v24 was uploaded to apk mirror (see https://www.apkmirror.com/apk/samsung-electronics-co-ltd/samsung-internet-for-android/samsung-internet-for-android-24-0-0-47-release/)

**Comment by Elchi3** [comment]
@matatk can you review and let us know when Samsung Internet 24.0 came out?

File: browsers/samsunginternet_android.json (hunk 1 above)
Comment (Elchi3) [reply browsers/samsunginternet_android.json:281]: Thanks for the link. iIt seems that version 25 was released yesterday?

Comment (lukewarlow) [reply browsers/samsunginternet_android.json:281]: Yep looks like it! I'll update this PR

**Review by Elchi3 — APPROVED** [review]
I agree, thanks @lukewarlow

**Pull request merged**